cmd/pipeline/main.go

```diff
@@ -587,10 +586,34 @@ func main() {
 
 	engine.Use(cors.New(corsConfig))
 
-	if config.Audit.Enabled {
+	if config.AuditLog.Enabled {
 		logger.Info("Audit enabled, installing Gin audit middleware")
-		engine.Use(audit.LogWriter(skipPaths, config.Audit.Headers, db, logrusLogger))
+
+		driver := auditlog.Drivers{}
+
+		if config.AuditLog.Driver.Log.Enabled {
+			driver = append(driver, auditlogdriver.NewLogDriver(
+				config.AuditLog.Driver.Log.Config,
+				logur.WithField(logger, "channel", "audit"),
+			))
+		}
+
+		if config.AuditLog.Driver.Database.Enabled {
+			driver = append(driver, auditlogdriver.NewDatabaseDriver(db))
+		}
+
+		engine.Use(auditlog.Middleware(
+			driver,
+			auditlog.WithUserIDExtractor(auth.GetCurrentUserID),
+			auditlog.WithSensitivePaths([]*regexp.Regexp{
+				regexp.MustCompile("/auth/dex(?:/.*)?"),
+				regexp.MustCompile("/.*/api/v1/orgs/.*/secrets(?:/.*)?"),
+				regexp.MustCompile("/.*/api/v1/orgs/.*/clusters/.*/pke/ready"),
```

Comment:
making these configurable by the user (with these sane defaults)?

Reply:
There are certain URLs that have to be filtered (body not logged). But I don't see a use case for the user specifying these URLs. Do you have one in mind?

Reply:
I had this in mind: having an issue where the outcome is to instantly add more urls to this list, or simply to fix one of them, where otherwise you would need a new build instead of a reconfiguration.